Release notes for v5.0.7 and v4.6.29 (#3133)
ezsystems/developer-documentation: docs/release_notes/ibexa_dxp_v5.0.md

<div class="release-notes" markdown="1">

[[% set version = 'v5.0.7' %]]

[[= release_note_entry_begin(
    "Google Gemini connector " + version,
    '2026-04-14',
    ['Headless', 'Experience', 'Commerce', 'LTS Update', 'New feature', 'First release']
) =]]

This release introduces a new AI connector that allows you to integrate [AI Actions](https://doc.ibexa.co/en/5.0/ai_actions/ai_actions/) with [Google Gemini](https://gemini.google/overview/#what-gemini-is).

> You can use it as an alternative embeddings provider for the [taxonomy suggestions feature](taxonomy.md#taxonomy-suggestions).

Reviewer: ```suggestion
You can also use it as an alternative embeddings provider for the [taxonomy suggestions feature](taxonomy.md#taxonomy-suggestions).
```
This makes it sounds like it's the only use case

#### Tracking

This add-on includes two Twig functions to ease tracking setting:

- `ibexa_tracking_script` to load the JavaScript tracking code, for more information, see [Tracking script](https://doc.ibexa.co/en/5.0/cdp/raptor_integration/tracking_script/)
- `ibexa_tracking_track_event` to send tracking events from your pages, for more information, see [Tracking event function](https://doc.ibexa.co/en/5.0/cdp/raptor_integration/tracking_event_function/)

#### Recommendations blocks [[% include 'snippets/experience_badge.md' %]] [[% include 'snippets/commerce_badge.md' %]]

This add-on introduces a set of recommendation blocks available in the [Page Builder](https://doc.ibexa.co/en/5.0/content_management/pages/page_builder_guide/), designed to suggest relevant content or products to users, such as the most popular items or viewed by others.

For more information about Recommendation blocks in Page Builder, see the relevant [Developer Documentation](https://doc.ibexa.co/en/5.0/cdp/raptor_integration/pb_blocks/) and [User Documentation](https://doc.ibexa.co/projects/userguide/en/5.0/recommendations/raptor_integration/raptor_recommendation_blocks/).

### Quable PIM

> This add-on enables integration between [[[= pim_product_name =]]](https://www.quable.com/en) and [[= product_name =]].

Reviewer: ```suggestion
The [[= pim_product_name =]] integration add-on allows you to connect [[= product_name =]] with [[[= pim_product_name =]] Product Information Management (PIM)](https://www.quable.com/en), making [[= pim_product_name =]] the authoritative source of product information for every website powered by [[= product_name =]].

[[= pim_product_name =]] can serve as the single source of truth for all product data, including attributes, classifications, variants, and translations. 
[[= product_name =]] consumes this data and makes it available for use in content and digital experiences.
```

For more information, see [Quable PIM Integration](https://doc.ibexa.co/en/5.0/product_catalog/quable/).

### Developer experience

#### Symfony 7.4

> Symfony is upgraded from 7.3 to 7.4.

Reviewer: ```suggestion
Symfony is upgraded from 7.3 to 7.4.
It's the latest [LTS release](https://symfony.com/releases#long-term-support-release), maintainted till November 2029.
```

See [what's new in Symfony 7.4](https://symfony.com/blog/category/living-on-the-edge/8.0-7.4) and [how to update Symfony within [[= product_name =]]](https://doc.ibexa.co/en/5.0/update_and_migration/from_5.0/update_from_5.0/#update-symfony-from-73-to-74).

#### Taxonomy search

> Two [taxonomy search](https://doc.ibexa.co/en/5.0/content_management/taxonomy/taxonomy_api/search) criteria are added:

Reviewer: To be completely honest, TaxonomySubtree was added 3 years ago and missed 🙈 But I think it's ok to mention it here (https://github.com/ibexa/documentation-developer/pull/3082#pullrequestreview-3929135160)

Author: I could mention one here and keep the other for new_in_doc.md

    'method': 'inline',
    'params': {
        'custom_param': 'custom_value',
        'another_param': 'another_value'
    }
}) }}
```

The parameters are available in your template as regular variables.

For more information, see [`ibexa_render()` Twig function](https://doc.ibexa.co/en/5.0/templating/twig_function_reference/content_twig_functions/#ibexa_render).

#### Try-catch support in data migrations

Data migrations now support try-catch error handling, allowing you to wrap migration steps with exception handling logic.
You can use it for migrations that might fail under certain conditions but should not break the entire migration process.

For example, you can create languages without checking if they already exist:

``` yaml
[[= include_file('code_samples/data_migration/examples/try_catch_step.yaml') =]]
```

The `try_catch` step allows you to specify which exceptions to catch and whether to continue executing remaining steps after an exception occurs.

For more information, see [Error handling with try-catch](https://doc.ibexa.co/en/5.0/content_management/data_migration/importing_data/#error-handling-with-try-catch).

#### Translation-related Twig Component groups

> Four groups of [Twig components](https://doc.ibexa.co/en/5.0/templating/components/) related to Admin UI translation are added.

Reviewer: ```suggestion
Four new [Twig component groups](https://doc.ibexa.co/en/5.0/templating/components/) related to Admin UI translation are added:

- `admin-ui-product-translation-modal-footer`
- `admin-ui-product-translations-actions-modal`
- `admin-ui-product-translations-actions`
- `admin-ui-product-translations-row-actions`
```

> For more information, see [Customizing the back office with Twig Components > Admin UI](https://doc.ibexa.co/en/5.0/administration/back_office/back_office_elements/custom_components/#admin-ui).

Reviewer: ```suggestion
For more information, see [available Admin UI Twig Component groups](https://doc.ibexa.co/en/5.0/administration/back_office/back_office_elements/custom_components/#admin-ui).
```

#### REST API

You can now find examples for some REST request bodies in the [OpenAPI REST API](rest_api_usage.md#openapi-support):

> - in the right column of the [online reference](https://doc.ibexa.co/en/5.0/api/rest_api/rest_api_reference/rest_api_reference.html) where the example is used as a sample instead of a duplicate of middle column schema,

Reviewer: ```suggestion
- in the right column of the [online reference](https://doc.ibexa.co/en/5.0/api/rest_api/rest_api_reference/rest_api_reference.html),
```
I wouldn't go that deep into what has changed

- [`Ibexa\Contracts\ProductCatalog\Events\ProductAttributeRenderEvent`](https://doc.ibexa.co/en/5.0/api/php_api/php_api_reference/classes/Ibexa-Contracts-ProductCatalog-Events-ProductAttributeRenderEvent.html)
- [`Ibexa\Contracts\Taxonomy\Search\Query\Criterion\TaxonomyNoEntries`](https://doc.ibexa.co/en/5.0/api/php_api/php_api_reference/classes/Ibexa-Contracts-Taxonomy-Search-Query-Criterion-TaxonomyNoEntries.html)
  <br>For more information, see [search criteria reference entry](https://doc.ibexa.co/en/5.0/search/criteria_reference/taxonomy_no_entries/).
- [`Ibexa\Contracts\ConnectorRaptor` namespace](https://doc.ibexa.co/en/5.0/api/php_api/php_api_reference/namespaces/ibexa-contracts-connectorraptor.html) from the [Raptor connector add-on](https://doc.ibexa.co/en/5.0/cdp/raptor_integration/raptor_connector/)
- [`Ibexa\Contracts\IntegratedHelp` namespace](https://doc.ibexa.co/en/5.0/api/php_api/php_api_reference/namespaces/ibexa-contracts-integratedhelp.html) from the [Integrated help LTS-Update](https://doc.ibexa.co/en/5.0/administration/back_office/integrated_help/)

> ### AI Actions in Page Builder blocks [[% include 'snippets/experience_badge.md' %]] [[% include 'snippets/commerce_badge.md' %]]

Reviewer: I would move above the Developer experience - it's a feature, albeit small

Author: It was there because of its scope EC but I'm agree
Done in b35b131e613113e4e4ce1d486aa89e0df1e2025b

<div class="release-notes" markdown="1">

[[% set version = 'v5.0.7' %]]

[[= release_note_entry_begin(
    "Google Gemini connector " + version,
    '2026-04-14',
    ['Headless', 'Experience', 'Commerce', 'LTS Update', 'New feature', 'First release']
) =]]

This release introduces a new AI connector that allows you to integrate [AI Actions](https://doc.ibexa.co/en/5.0/ai_actions/ai_actions/) with [Google Gemini](https://gemini.google/overview/#what-gemini-is).

> Among other things, you can use it as an alternative embeddings provider for the taxonomy suggestions feature.

Reviewer 2: Link could be a nice addition (please double check if it's correct).
Also, "_Among other things_" can cause questions "What other things?", I'd skipp it.
```suggestion
You can use it as an alternative embeddings provider for the [taxonomy suggestions feature](taxonomy.md#taxonomy-suggestions).
```

For more information, see how to [install and configure the Google Gemini connector](https://doc.ibexa.co/en/5.0/ai_actions/configure_ai_actions/#install-google-gemini-connector).

[[= release_note_entry_end() =]]

[[= release_note_entry_begin(
    "Integrated help " + version,
    '2026-04-14',
    ['Headless', 'Experience', 'Commerce', 'LTS Update', 'New feature']
) =]]

### Product tour

The product tour is a new Integrated help feature that helps back office contributors to discover [[= product_name =]].

For more information, see [Product tour](https://doc.ibexa.co/en/5.0/administration/back_office/product_tour/).

[[= release_note_entry_end() =]]

[[= release_note_entry_begin(
    "Ibexa DXP " + version,
    '2026-04-14',
    ['Headless', 'Experience', 'Commerce', 'New feature']
) =]]

### Raptor connector

> The SiteAccess-aware Raptor connector provides a seamless integration between [[= product_name =]] and [Raptor Recommendation Engine](https://www.raptorservices.com/website-recommendations/).

Reviewer 2: In feature-branch we've decided to skip SiteAccess-aware info.
```suggestion
The Raptor connector provides a seamless integration between [[= product_name =]] and [Raptor Recommendation Engine](https://www.raptorservices.com/website-recommendations/).
```

> For more information, see [SiteAccess-aware Raptor connector](https://doc.ibexa.co/en/5.0/cdp/raptor_integration/raptor_connector/).

Reviewer 2: ```suggestion
For more information, see [Raptor connector](https://doc.ibexa.co/en/5.0/cdp/raptor_integration/raptor_connector/).
```

#### Tracking

This add-on includes two Twig functions to ease tracking setting:

- `ibexa_tracking_script` to load the JavaScript tracking code, for more information, see [Tracking script](https://doc.ibexa.co/en/5.0/cdp/raptor_integration/tracking_script/)
- `ibexa_tracking_track_event` to send tracking events from your pages, for more information, see [Tracking event function](https://doc.ibexa.co/en/5.0/cdp/raptor_integration/tracking_event_function/)

#### Recommendations blocks [[% include 'snippets/experience_badge.md' %]] [[% include 'snippets/commerce_badge.md' %]]

This add-on introduces a set of recommendation blocks available in the [Page Builder](https://doc.ibexa.co/en/5.0/content_management/pages/page_builder_guide/),

> designed to suggest relevant content or products to users, such as the most popular items or viewed by others.

Reviewer 2: ```suggestion
This add-on introduces a set of recommendation blocks available in the [Page Builder](https://doc.ibexa.co/en/5.0/content_management/pages/page_builder_guide/), designed to suggest relevant content or products to users, such as the most popular items or viewed by others.
```

> For more information, see [Recommendation blocks in Page Builder](https://doc.ibexa.co/en/5.0/cdp/raptor_integration/pb_blocks/).

Reviewer 2: What about linking to user doc as well?
(please double check the link)
```suggestion
For more information about Recommendation blocks in Page Builder, see the relevant [Developer Documentation](https://doc.ibexa.co/en/5.0/cdp/raptor_integration/pb_blocks/) and [User Documentation](https://doc.ibexa.co/projects/userguide/en/5.0/recommendations/raptor_integration/raptor_recommendation_blocks/).
```

### Quable PIM

> Add-on to integrate [[[= pim_product_name =]]](https://www.quable.com/en) with [[= product_name =]].

Reviewer 2: ```suggestion
This add-on enables integration between [[[= pim_product_name =]]](https://www.quable.com/en) and [[= product_name =]].
```

For more information, see [Quable PIM Integration](https://doc.ibexa.co/en/5.0/product_catalog/quable/).

### Developer experience

#### Symfony 7.4

Symfony is upgraded from 7.3 to 7.4.
See [what's new in Symfony 7.4](https://symfony.com/blog/category/living-on-the-edge/8.0-7.4)

> and [how to update Symfony within [[= product_name =]]](https://doc.ibexa.co/en/5.0/update_and_migration/from_5.0/update_from_5.0/#update-symfony-from-73-to-74).

Reviewer 2: ```suggestion
See [what's new in Symfony 7.4](https://symfony.com/blog/category/living-on-the-edge/8.0-7.4) and [how to update Symfony within [[= product_name =]]](https://doc.ibexa.co/en/5.0/update_and_migration/from_5.0/update_from_5.0/#update-symfony-from-73-to-74).
```

#### Taxonomy search

Two [taxonomy search](https://doc.ibexa.co/en/5.0/content_management/taxonomy/taxonomy_api/search) criteria are added:

- [`TaxonomyNoEntries`](https://doc.ibexa.co/en/5.0/search/criteria_reference/taxonomy_no_entries/) to find content items to which no taxonomy entries have been assigned.
- [`TaxonomySubtree`](https://doc.ibexa.co/en/5.0/search/criteria_reference/taxonomy_subtree/) for content items assigned to a taxonomy entry or any of its descendants.

#### Custom parameters in `ibexa_render()`

> You can now pass custom parameters to templates when using the `ibexa_render()` Twig function with the new `params` option, similar to how you can with `render(controller())`.Collapse annotationCheck notice on line R18[vale] docs/release_notes/ibexa_dxp_v5.0.md#L18Check notice: [vale] docs/release_notes/ibexa_dxp_v5.0.md#L18[Ibexa.ByUsing] Prefer 'by using' or 'with' to plain 'using'.Build & test documentation / valeView details

Reviewer 2: Same as in 4.6

#### Tracking

This add-on includes two Twig functions to ease tracking setting:

- `ibexa_tracking_script` to load the JavaScript tracking code, for more information, see [Tracking script](https://doc.ibexa.co/en/5.0/cdp/raptor_integration/tracking_script/)
- `ibexa_tracking_track_event` to send tracking events from your pages, for more information, see [Tracking event function](https://doc.ibexa.co/en/5.0/cdp/raptor_integration/tracking_event_function/)

#### Recommendations blocks [[% include 'snippets/experience_badge.md' %]] [[% include 'snippets/commerce_badge.md' %]]

This add-on introduces a set of recommendation blocks available in the [Page Builder](https://doc.ibexa.co/en/5.0/content_management/pages/page_builder_guide/), designed to suggest relevant content or products to users, such as the most popular items or viewed by others.

For more information about Recommendation blocks in Page Builder, see the relevant [Developer Documentation](https://doc.ibexa.co/en/5.0/cdp/raptor_integration/pb_blocks/) and [User Documentation](https://doc.ibexa.co/projects/userguide/en/5.0/recommendations/raptor_integration/raptor_recommendation_blocks/).

> ### Quable PIM

Reviewer: ```suggestion
### [[= pim_product_name =]] PIM
```
Vale is right 😄

The [[= pim_product_name =]] integration add-on allows you to connect [[= product_name =]] with [[[= pim_product_name =]] Product Information Management (PIM)](https://www.quable.com/en), making [[= pim_product_name =]] the authoritative source of product information for every website powered by [[= product_name =]].

[[= pim_product_name =]] can serve as the single source of truth for all product data, including attributes, classifications, variants, and translations. 
[[= product_name =]] consumes this data and makes it available for use in content and digital experiences.

For more information, see [Quable PIM Integration](https://doc.ibexa.co/en/5.0/product_catalog/quable/).

### AI Actions in Page Builder blocks [[% include 'snippets/experience_badge.md' %]] [[% include 'snippets/commerce_badge.md' %]]

You can now use the [refining text AI Actions](https://doc.ibexa.co/en/5.0/ai_actions/ai_actions_guide/#refining-text) in Page Builder blocks string and text inputs.

### Developer experience

#### Symfony 7.4

Symfony is upgraded from 7.3 to 7.4.

> It's the latest [LTS release](https://symfony.com/releases#long-term-support-release), maintainted till November 2029.

Reviewer: ```suggestion
It's the latest [LTS release](https://symfony.com/releases#long-term-support-release), maintained till November 2029.
```
Sorry for wrong suggestion 🙈
